internal/provider/data_source_pullzone.go

```diff
@@ -0,0 +1,86 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourcePullZone() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourcePullZoneRead,
+
+		Schema: map[string]*schema.Schema{
+			"pull_zone_id": {
```

Comment:
move `pull_zone_id` to a `keyPullZoneID` constant, as we do it for all other keys